yax/modules/identify_informative_hits.py

```diff
@@ -0,0 +1,217 @@
+from yax.shared.utilities import taxidtool
+from yax.artifacts import TaxIDData, InformativeAlignmentData, IdentifiedTaxa
+from yax.state.type.parameters import Int, File
+
+
+def main(working_dir, output, details, identified_taxa_art: IdentifiedTaxa,
+         tax_id_data_art: TaxIDData, lca_dist: Int, min_hits: Int,
+         min_relative_hits: Int, sequences_input_fp: File) \
+            -> InformativeAlignmentData:
+    """Produces InformativeAlignmentData artifact type
+
+    The InformativeAlignmentData artifact contains a FASTA formatted file and a
+    file representing the tree observed when built from alignments. The hits in
+    these alignments are first determined to be informative or not based on
+    least common ancestor and minimum hits on tax id to be informative.
+
+    Parameters
+    ----------
+        identified_taxa_art : IdentifiedTaxa
+            artifact from an alignment step that describes reads that hit to
+            tax ids
+        tax_id_data_art : TaxIDData
+            tax id data file artifact
+        lca_dist: Int
+            edges to traverse between multiple tax ids when determining if a
+            read that hit multiple taxids is informative
+        min_hits: Int
+            absolute number of hits a taxid must exhibit to be determined
+            informative
+        min_relative_hits: Int
+            number of hits relative to total number of hits reported that a
+            taxid must exhibit to be determined informative
+        sequences_input_fp: File
+            FASTA formatted file containing sequence data based on gi
+
+    Returns
+    -------
+        output
+            generates InformativeAlignmentData artifact
+
+    """
+    hit_data = output
+
+    hit_data.sequences_input_fp = sequences_input_fp
+    hit_data.truncation_level = "species"
+    hit_data.tax_id_data = taxidtool.parse_taxid_data(tax_id_data_art)
+
+    with open(identified_taxa_art, mode='r') as fh:
```

Comment:
You can also create methods on the artifacts if they are useful. I notice you use properties almost exclusively, there's no reason you can't create getters/setters where appropriate.